components/skus/browser/skus_sdk_context.cc

```diff
@@ -0,0 +1,121 @@
+#include "services/preferences/public/cpp/dictionary_value_update.h"
+#include "services/preferences/public/cpp/scoped_pref_update.h"
+
+namespace {
+
+void OnScheduleWakeup(
+    rust::cxxbridge1::Fn<
+        void(rust::cxxbridge1::Box<brave_rewards::WakeupContext>)> done,
+    rust::cxxbridge1::Box<brave_rewards::WakeupContext> ctx) {
+  done(std::move(ctx));
+}
+
+}  // namespace
+
+namespace brave_rewards {
+
+void shim_purge(brave_rewards::SkusSdkContext& ctx) {
+  ctx.PurgeStore();
+}
+
+void shim_set(brave_rewards::SkusSdkContext& ctx,
+              rust::cxxbridge1::Str key,
+              rust::cxxbridge1::Str value) {
+  ctx.UpdateStoreValue(static_cast<std::string>(key),
+                       static_cast<std::string>(value));
+}
+
+::rust::String shim_get(brave_rewards::SkusSdkContext& ctx,
+                        rust::cxxbridge1::Str key) {
+  return ::rust::String(ctx.GetValueFromStore(static_cast<std::string>(key)));
+}
+
+void shim_scheduleWakeup(
+    ::std::uint64_t delay_ms,
+    rust::cxxbridge1::Fn<
+        void(rust::cxxbridge1::Box<brave_rewards::WakeupContext>)> done,
+    rust::cxxbridge1::Box<brave_rewards::WakeupContext> ctx) {
+  LOG(ERROR) << "shim_scheduleWakeup " << delay_ms;
+  base::SequencedTaskRunnerHandle::Get()->PostDelayedTask(
+      FROM_HERE,
+      base::BindOnce(&OnScheduleWakeup, std::move(done), std::move(ctx)),
+      base::TimeDelta::FromMilliseconds(delay_ms));
+}
+
+std::unique_ptr<SkusSdkFetcher> shim_executeRequest(
+    const brave_rewards::SkusSdkContext& ctx,
+    const brave_rewards::HttpRequest& req,
+    rust::cxxbridge1::Fn<
+        void(rust::cxxbridge1::Box<brave_rewards::HttpRoundtripContext>,
+             brave_rewards::HttpResponse)> done,
+    rust::cxxbridge1::Box<brave_rewards::HttpRoundtripContext> rt_ctx) {
+  auto fetcher = ctx.CreateFetcher();
+  fetcher->BeginFetch(req, std::move(done), std::move(rt_ctx));
+  return fetcher;
+}
+
+// static
+void SkusSdkContext::RegisterProfilePrefs(
+    user_prefs::PrefRegistrySyncable* registry) {
+  registry->RegisterDictionaryPref(prefs::kSkusDictionary);
+  registry->RegisterStringPref(prefs::kSkusVPNCredential, "");
+}
+
+SkusSdkContext::SkusSdkContext(
+    PrefService* prefs,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : prefs_(prefs), url_loader_factory_(url_loader_factory) {}
+
+SkusSdkContext::~SkusSdkContext() {}
+
+std::unique_ptr<brave_rewards::SkusSdkFetcher> SkusSdkContext::CreateFetcher()
+    const {
+  return std::make_unique<SkusSdkFetcher>(url_loader_factory_);
+}
+
+std::string SkusSdkContext::GetValueFromStore(std::string key) const {
+  LOG(ERROR) << "shim_get: `" << key << "`";
+  const base::Value* dictionary = prefs_->GetDictionary(prefs::kSkusDictionary);
+  DCHECK(dictionary);
+  DCHECK(dictionary->is_dict());
+  const base::Value* value = dictionary->FindKey(key);
+  if (value) {
+    return value->GetString();
+  }
+  return "{}";
```

Comment:
it would be better to return `absl::optional<std::string>`

Reply:
Updated - had to return empty string (ran into a problem here)